### PR TITLE
Fix invisible GPX tracks after the import

### LINKED ISSUE #5800: GPX imports not visible on map
**Describe the issue**
GPX imports don't seem to work properly in v2023.08.18-7 (the current App Store release). A success message is displayed, but the imported trail is not visible on the map.

**Steps to reproduce**
1. Obtain a GPX file (example from AllTrails attached below).
2. Tap the share/open button in iOS and select OrganicMaps from the list of apps.
3. "Bookmarks loaded successfully!" is displayed.
4. The trail is visible and ticked in the "Bookmarks and Tracks" list.
5. It is missing/invisible on the map itself.

Example GPX file:
[Valley of the Waters to Wentworth Falls.gpx.zip](https://github.com/organicmaps/organicmaps/files/12418226/Valley.of.the.Waters.to.Wentworth.Falls.gpx.zip)

Corresponding KMZ file, which does import & display properly:
[Valley of the Waters to Wentworth Falls.kmz.zip](https://github.com/organicmaps/organicmaps/files/12418296/Valley.of.the.Waters.to.Wentworth.Falls.kmz.zip)

**Expected behavior**
The imported trail should be visible on the map (or at least, an explicit import error should occur).

**Screenshots**
1. Import appears to complete successfully:
IMG_0113

2. Imported trail is visible & ticked in this list:
IMG_0115

3. It is missing from the map view:
IMG_0114

4. If I repeat the process with the KMZ file instead, it displays correctly:
IMG_0117

**System information:**
 - Operating system and its version: iOS 16.5.1 (c)
 - Organic Maps version: 2023.08.18-7 (OSM data: 14 August 2023)
 - Device Model: iPhone XS Max

**Additional context**

- Same behaviour was observed with a GPX file from a different source (Wikiloc), so I don't think it's just an AllTrails thing.
- It worked correctly for me with the previous App Store release. 
- A GPX trail I’d imported using the previous release is still displaying correctly, it’s just new imports that don’t work.

### PR DESCRIPTION
**Pull request by biodranik**
Fixes #5800
Supersedes and closes #5824

@cyber-toad

### CODE REVIEW

File: coding/coding_tests/mem_file_reader_test.cpp [hunk 1]
```diff
@@ -24,3 +24,25 @@ UNIT_TEST(MemReaderSimple)
...
+    uint8_t c[n] = {0};
+    ReadFromPos(memReader, i, c, n - i);
+    for (size_t j = 0; j < n; ++j)
+      TEST_EQUAL(c[j], i + j < n ? data[i + j] : uint8_t{0}, (i, j, n));
+  }
+  MemReader const subReader = memReader.SubReader(1, 2);
+  for (size_t i = 1; i < n - 1; ++i)
+  {
+    uint8_t c[n] = {0};
+    ReadFromPos(subReader, i, c, n - i - 2);
```
Comment (vng) [inline coding/coding_tests/mem_file_reader_test.cpp:44]: Strange test, honestly saying ..
i == 1 only
size == 3-1-2 == 0
subReader from i == 1 - is its end

File: kml/serdes.cpp [hunk 2]
```diff
@@ -984,9 +976,13 @@ void KmlParser::Pop(std::string_view tag)
     // loading of KML files which were stored by older versions of OMaps.
     TrackLayer layer;
     layer.m_lineWidth = m_trackWidth;
-    layer.m_color.m_predefinedColor = PredefinedColor::None;
-    layer.m_color.m_rgba = (m_color != 0 ? m_color : kDefaultTrackColor);
-    m_trackLayers.push_back(std::move(layer));
+    // Fix wrongly parsed transparent color, see https://github.com/organicmaps/organicmaps/issues/5800
+    // TODO: Remove this fix in 2024 when all users will have their imported GPX files fixed.
+    if (m_color == 0 || (m_color & 0xFF) < 10)
```
Comment (vng) [inline kml/serdes.cpp:981]: Is it possible that unitialized value was random? Do we really need it or users can just re-import their tracks?

File: coding/coding_tests/mem_file_reader_test.cpp (hunk 1 above)
Comment (biodranik) [reply coding/coding_tests/mem_file_reader_test.cpp:44]: Fixed.

File: kml/serdes.cpp (hunk 2 above)
Comment (biodranik) [reply kml/serdes.cpp:981]: In theory, everything is possible. A random non-transparent value means the track is visible, so it is OK.
Asking users to re-import is not a good idea. Do you see any risks in this fix? We can remove it after a few releases.

**Review by vng — APPROVED** [review]

**Comment by vng** [comment]
Add somewhere to Gpx_Test_Route checking default color and width:
```
auto const & layer = dataFromText.m_trackData[0].m_trackLayer[0];
TEST_EQUAL(layer.m_color, ...);
TEST_EQUAL(layer.m_lineWidth, ...)
```

**Comment by vng** [comment]
Check default color in gpx_test.cpp
Actually, where the bug was.
I wrote you a test name above.

**Review by vng — APPROVED** [review]

**Pull request merged**